src/lua/protobuf.lua

```diff
@@ -500,6 +523,19 @@ local function validate_type(field_def, value, exp_type)
     return
 end
 
+local function validate_field_id(field_id)
+    if field_id < MIN_FIELD_ID or field_id > MAX_FIELD_ID then
+        error(('Id %d is out of range [%d; %d]'):format(
+            field_id, MIN_FIELD_ID, MAX_FIELD_ID))
+    end
+    if field_id >= RESERVED_FIELD_ID_MIN and
+        field_id <= RESERVED_FIELD_ID_MAX then
+        error(('Id %d is in reserved ' ..
+            'id range [%d, %d]'):format(field_id,
+            RESERVED_FIELD_ID_MIN, RESERVED_FIELD_ID_MAX))
+    end
+end
```

Comment:
A very similar code is in `protobuf.message()`. Can we reuse the function there?

Reply:
We can use this function but there is one problem. If this error raises on encoding we know `field_name` from protocol and can use it in error message. On decoding we can't know `field_name` because `field_id` is incorrect. I adjusted error message for this situation, now it returnes `decoded` instead of name if an error occures while decoding. I suppose that I can add this logic, that will change error message, in upcoming `wrap_error` function.

Reply:
It seems OK to adjust the error message depending on whether there is a known field name.